src/main/java/cn/nukkit/entity/item/EntityFallingBlock.java

```diff
@@ -175,6 +181,13 @@ public boolean onUpdate(int currentTick) {
 
                         if (event.getTo().getId() == Item.ANVIL) {
                             getLevel().addLevelEvent(block, LevelEventPacket.EVENT_SOUND_ANVIL_FALL);
+
+                            Entity[] e = level.getCollidingEntities(this.getBoundingBox(), this);
+                            for (Entity entity : e) {
+                                if (entity instanceof EntityLiving) {
+                                    entity.attack(new EntityDamageByBlockEvent(event.getTo(), entity, DamageCause.CONTACT, (float) Math.min(40, Math.max(0, (startY - y) * 2))));
```

Comment:
Wouldn't startY - fall distance be resetted with every entity reload?

Reply:
Yes

Reply:
It might be better to just use fallDistance that is managed by Entity class